### PR TITLE
Adjust heatmap function

### PR DESCRIPTION
**Pull request by mducducd**
Adjust STAMP gradient function as follows:
- Old: logits -> softmax -> compute gradients (may cause vanishing signal)
- New: logits -> compute gradients -> softmax
This adjustment is more consistent with the method described in the Grad-CAM paper. As a result, the colors across slides now appear more consistent, and each heatmap has clearer, high-contrast regions.

### CODE REVIEW

**Review by EzicStar — APPROVED** [review]
LGTM!
